### PR TITLE
Eslint

### PR DESCRIPTION
**Pull request by yeyu456**
Convert JSHint && JSCS rule to eslint with [polyjuice](https://github.com/brenolf/polyjuice).
And manually do some correction.
JSHint rules are totally compatible, JSCS ones are not that lucky.
The rules isn't compatible had been commented with TODO label.

### CODE REVIEW

**Comment by JSteunou** [comment]
Nice tool, I did not know about it!

**Comment by JSteunou** [comment]
Ok I checked and I have still some issue with eslint. The major one is related to spaces, and how it makes no differences for spaces inside function signature and outside. And issue is opened about that since a while now :( https://github.com/eslint/eslint/issues/3587

Maybe you'll got a solution about this.

In the mean time could you update your PR to add eslint into the package.json, remote jscs & jshint, and update the `lint` scripts entry?

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        "no-undef": 2,
+
+        // Warns when variables are defined but never used.
+        "no-unused-vars": 2,
+
+        // requireCurlyBraces
+        // TODO Not completely compatible with original rule.
+        "curly": [
+            2,
+            "all"
```
Comment (JSteunou) [inline .eslintrc]: You can remote the TODO comment (in fact all the TODO comments) and replace `all` with `"multi-line", "consistent"`

**Comment by yeyu456** [comment]
> In the mean time could you update your PR to add eslint into the package.json, remote jscs & jshint, and update the lint scripts > 

Sure, give me some time i will finish it.

>  And issue is opened about that since a while now :( eslint/eslint#3587

Hahaha, such a long time, sigh.Maybe we need a PR :).

**Comment by yeyu456** [comment]
As your wish, done.:laughing:

**Comment by JSteunou** [comment]
I could not take your PR, too much changes on client.js + some tab / space weirdness elsewhere so I cherry pick some and re-write some part. thank you @yeyu456 this project is now eslint!

**Pull request closed without merging**